crates/frontend/src/components/default_config_form.rs

```diff
@@ -150,12 +150,18 @@ where
             <Suspense>
                 {move || {
                     let options = type_template_resource.get().unwrap_or(vec![]);
-                    let config_t = if config_type_rs.get().is_empty() {
+                    let config_t = if config_type_rs.get().is_empty() && edit {
+                        "change current type template".into()
```

Comment:
Why are we allowing the default config type to change ?
Wouldn't that be an issue for existing overrides , just wanted to understand the use case where we would need it.

Reply:
@sauraww adding an enum, changing the pattern - these are few examples where we would want to update the type. I replicated what was the behaviour today. If we want to stop this, we discuss and do it in another PR

Reply:
@sauraww this functionality of changing the schema of a default_config is already there. And the change would effect any new overrides i think, it wouldn't effect the existing ones i guess.